go/home/userdata.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package home
+
+import (
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/phonenumbers"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+func getUserDataForItem(mctx libkb.MetaContext, item *keybase1.HomeScreenItem) error {
+	todoItem := item.Data.Todo()
+	// We can't do todoItem.T() here because it would return an error because
+	// variants we are getting from the server are incomplete. We are expected
+	// to fill them here in this function. Also we want to be modifying
+	// existing HomeScreenTodo objects in place, so we can't use accessor
+	// methods.
+	todoType := todoItem.T__
+	switch todoType {
+	case keybase1.HomeScreenTodoType_VERIFY_ALL_PHONE_NUMBER:
```

Comment:
what happens if there is a mismatch between the server sending a TODO but not finding any data to attach to the TODO? Will something else fail with an unexpected nil on the variant? perhaps this function should signal the caller if no data is added so the todo item can be skipped

Reply:
you're right, this is not handled gracefully at all.

here is what I can do - allow this to be nil when we can't find data, and have the frontend be able to handle it if it is. Rather than e.g. showing `undefined` in the todo description instead of phone number.

Reply:
why not just hide the TODO?

Reply:
Sounds good. But server only gives us two at a time so we will not have a todo item to fall back on if this happens. Is this ok?

Reply:
this seems like a bad place to do this stuff to, because this goes to the cache